src/Features/CSharp/Portable/OrderModifiers/CSharpOrderModifiersHelper.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.CSharp.OrderModifiers
+{
+    internal static class CSharpOrderModifiersHelper
+    {
+        private static readonly char[] s_comma = { ',' };
+
+        /// <remarks>
+        /// Reference type so we can read/write atomically.
+        /// </remarks>
+        private static Tuple<string, Dictionary<int, int>> s_lastParsed;
+
+        public static bool TryGetOrComputePreferredOrder(string value, out Dictionary<int, int> preferredOrder)
+        {
+            if (string.IsNullOrWhiteSpace(value))
+            {
+                preferredOrder = null;
+                return false;
+            }
+
+            var lastParsed = Volatile.Read(ref s_lastParsed);
+            if (lastParsed?.Item1 != value)
+            {
+                if (!TryParse(value, out var parsed))
+                {
+                    preferredOrder = null;
+                    return false;
+                }
+
+                lastParsed = Tuple.Create(value, parsed);
+                Volatile.Write(ref s_lastParsed, lastParsed);
+            }
+
+            preferredOrder = lastParsed.Item2;
+            return true;
+        }
+
+        private static bool TryParse(string value, out Dictionary<int, int> parsed)
+        {
+            var result = new Dictionary<int, int>();
+
+            var index = 0;
+            foreach (var piece in value.Split(s_comma, StringSplitOptions.RemoveEmptyEntries))
+            {
+                var trimmed = piece.Trim();
```

Comment:
:memo: This handles `"private,,protected"` and `"private, ,protected"` very differently.